src/EA.Iws.DataAccess/Repositories/MovementRepository.cs

```diff
@@ -99,5 +99,30 @@ public void Add(Movement movement)
 
             return movement == null ? 0 : movement.Number;
         }
+
+        public async Task<bool> DeleteById(Guid movementId)
+        {
+            try
+            {
+                await context.Database.ExecuteSqlCommandAsync(
+                @"DELETE from [Notification].[MovementCarrier] WHERE MovementDetailsId in (SELECT Id FROM [Notification].[MovementDetails] WHERE MovementId = @movementId)
+                  DELETE from [Notification].[MovementPackagingInfo] WHERE MovementDetailsId in (SELECT Id FROM [Notification].[MovementDetails] WHERE MovementId = @movementId)
+                  DELETE from [Notification].[MovementDetails] WHERE MovementId = @movementId
+                  DELETE from [Notification].[MovementDateHistory] WHERE MovementId = @movementId
+                  DELETE from [Notification].[MovementOperationReceipt] WHERE MovementId = @movementId
+                  DELETE from [Notification].[MovementReceipt] WHERE MovementId = @movementId
+                  DELETE from [Notification].[MovementRejection] WHERE MovementId = @movementId
+                  DELETE from [Notification].[MovementStatusChange] WHERE MovementId = @movementId
+                  DELETE from [Notification].[Movement] WHERE Id = @movementId",
+                new SqlParameter("@movementId", movementId));
+            }
+            catch (Exception e)
+            {
+                var bob = e;
+                return false;
+            }
```

Comment:
You can remove this try / catch (especially bob!).

The return value from ExecuteSqlCommandAsync is the number of rows effected **I think - double check this**. If so, you could use this, return true if > 1, false if == 0.